subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -1352,6 +1352,94 @@ static int cmd_wifi_ap_sta_disconnect(const struct shell *sh, size_t argc,
 	return 0;
 }
 
+static int __wifi_ap_config_args_to_params(const struct shell *sh, size_t argc, char *argv[],
+					   struct wifi_ap_config_params *params)
+{
+	struct getopt_state *state;
+	int opt;
+	static struct option long_options[] = {{"max_inactivity", required_argument, 0, 'i'},
+					       {"max_num_sta", required_argument, 0, 's'},
+					       {"help", no_argument, 0, 'h'},
+					       {0, 0, 0, 0}};
+	int opt_index = 0;
+	long val;
+	int err = 0;
+
+	while ((opt = getopt_long(argc, argv, "i:s:h", long_options, &opt_index)) != -1) {
+		state = getopt_state_get();
+		switch (opt) {
+		case 'i':
+			val = shell_strtol(optarg, 10, &err);
+			if (err) {
+				shell_error(sh, "Unable to parse input for max_inactivity (err %d)",
+					    err);
+				return err;
+			}
+
+			if (val < 0) {
+				PR_ERROR("Invalid value for max_inactivity %ld\n", val);
+				return -EINVAL;
+			}
+
+			params->max_inactivity = (uint32_t)val;
+			params->type |= WIFI_AP_CONFIG_PARAM_MAX_INACTIVITY;
+			break;
+		case 's':
+			val = shell_strtol(optarg, 10, &err);
+			if (err) {
+				shell_error(sh, "Unable to parse input for max_num_sta (err %d)",
+					    err);
+				return err;
+			}
+			if (val < 0) {
+				PR_ERROR("Invalid value for max_num_sta %ld", val);
+				return -EINVAL;
+			}
+			if (val > CONFIG_WIFI_MGMT_AP_MAX_NUM_STA) {
+				PR_ERROR("(%d): Exceeded system default configured value."
+					 " Increase the system default config\n",
+					 params->max_num_sta);
+				return -EINVAL;
+			}
+			params->max_num_sta = (uint32_t)val;
+			params->type |= WIFI_AP_CONFIG_PARAM_MAX_NUM_STA;
+			break;
+		case 'h':
+			return SHELL_CMD_HELP_PRINTED;
```

Comment:
This basically returns 1 and is typically used after printing `shell_help` see https://github.com/search?q=repo%3Azephyrproject-rtos%2Fzephyr%20SHELL_CMD_HELP_PRINTED&type=code 

and I see that the return value itself isn't used anywhere, just a convention to return non-zero value I guess.